docs/en/modules/mqtt.md

```diff
@@ -11,18 +11,39 @@ The client adheres to version 3.1.1 of the [MQTT](https://en.wikipedia.org/wiki/
 Creates a MQTT client.
 
 #### Syntax
-`mqtt.Client(clientid, keepalive[, username, password, cleansession])`
+`mqtt.Client(clientid, keepalive[, username, password, cleansession, max_message_length])`
 
 #### Parameters
 - `clientid` client ID
 - `keepalive` keepalive seconds
 - `username` user name
 - `password` user password
 - `cleansession` 0/1 for `false`/`true`. Default is 1 (`true`).
+- `max_message_length`, how large messages to accept. Default is 1024.
 
 #### Returns
 MQTT client
 
+#### Notes
+
+According to MQTT specification the max PUBLISH length is 256Mb. This is too large for NodeMCU to realistically handle. To avoid
+an out-of-memory situation, there is a limit on how big messages to accept. This is controlled by the `max_message_length` parameter.
+In practice, this only affects incoming PUBLISH messages since all regular control packets are small.
+The default 1024 was chosen as this was the implicit limit in NodeMCU 2.2.1 and older (where this was not handled at all).
+
+Any message *larger* than `max_message_length` will be (partially) delivered to the `overflow` callback, if defined. The rest
+of the message will be discarded. Any following messages should be handled properly.
+
+If raising this, heap memory will be used o buffer the received message data between each incoming TCP packet.
```

Comment:
"used to buffer"

But it's not just if the limit is raised, right?  It's whenever there might be a misalignment between the TCP and MQTT packetizations of the byte stream.  Ah, I see you address that below.  Maybe reword for clarity?

Reply:
Indeed! rewrote those two sentences, clearer now?

Reply:
Yes, thank you.